code/modules/spacepods/spacepod.dm

```diff
@@ -0,0 +1,1090 @@
+	var/datum/action/innate/spacepod/poddoor/door_action = new
+	var/datum/action/innate/spacepod/weapons/fire_action = new
+	var/datum/action/innate/spacepod/cargo/unload_action = new
+	var/datum/action/innate/spacepod/lights/light_action = new
+	var/datum/action/innate/spacepod/checkseat/seat_action = new
+	var/datum/action/innate/spacepod/airtank/tank_action = new
+
+	hud_possible = list(DIAG_HUD, DIAG_BATT_HUD)
+
+	var/datum/pod_armor/pod_armor
+
+
+/obj/spacepod/proc/apply_paint(mob/user as mob)
+	var/part_type
+	var/part = input(user, "Choose part", null) as null|anything in list("Lights","Rim","Paint","Windows")
+	switch(part)
+		if("Lights")
+			part_type = LIGHT
+		if("Rim")
+			part_type = RIM
+		if("Paint")
+			part_type = PAINT
+		if("Windows")
+			part_type = WINDOW
+		else
+	var/coloradd = input(user, "Choose a color", "Color") as color
+	colors[part_type] = coloradd
+	if(!has_paint)
+		has_paint = 1
+	update_icons()
+
+/obj/spacepod/proc/get_intergrity()
+	return obj_integrity/max_integrity*100
+
+/obj/spacepod/Initialize(var/mapload, var/datum/pod_armor/p_armor)
+	. = ..(mapload)
+	if (p_armor)
+		pod_armor = new p_armor
+
+	else
+		pod_armor = new
+	update_icons()
+	dir = EAST
+	cell = new cell_type(src)
+	add_cabin()
+	add_airtank()
+	ion_trail = new /datum/effect_system/trail_follow/ion/space_trail()
+	ion_trail.set_up(src)
+	ion_trail.start()
+	use_internal_tank = 1
+	GLOB.poi_list += src
+	equipment_system = new(src)
+	equipment_system.installed_modules += cell
+	GLOB.spacepods_list += src
+	START_PROCESSING(SSobj, src)
+	prepare_huds()
+	var/datum/atom_hud/data/diagnostic/diag_hud = GLOB.huds[DATA_HUD_DIAGNOSTIC]
+	diag_hud.add_to_hud(src)
+	diag_hud_set_podhealth()
+	diag_hud_set_podcharge()
+	max_integrity *= pod_armor.armor_multiplier
+	obj_integrity *= pod_armor.armor_multiplier
+	cargo_hold = new/obj/item/storage/internal(src)
+	cargo_hold.w_class = 5	//so you can put bags in
+	cargo_hold.storage_slots = 0	//You need to install cargo modules to use it.
+	cargo_hold.max_w_class = 5		//fit almost anything
+	cargo_hold.max_combined_w_class = 0 //you can optimize your stash with larger items
+
+/obj/spacepod/Destroy()
+	if(equipment_system.cargo_system)
+		equipment_system.cargo_system.removed(null)
+	QDEL_NULL(equipment_system)
+	QDEL_NULL(cargo_hold)
+	QDEL_NULL(cell)
+	STOP_PROCESSING(SSobj, src)
+	if(loc)
+		loc.assume_air(cabin_air)
+		air_update_turf()
+	QDEL_NULL(cabin_air)
+	QDEL_NULL(internal_tank)
+	QDEL_NULL(ion_trail)
+	occupant_sanity_check()
+	if(pilot)
+		pilot.forceMove(get_turf(src))
+		Remove_Actions(pilot)
+		pilot.clear_alert("charge")
+		pilot.clear_alert("mech damage")
+		pilot = null
+	if(passengers)
+		for(var/mob/living/M in passengers)
+			M.forceMove(get_turf(src))
+			Remove_Actions(M)
+			passengers -= M
+	GLOB.spacepods_list -= src
+	GLOB.poi_list.Remove(src)
+	return ..()
+
+/obj/spacepod/proc/update_icons()
+	bound_width = 64
+	bound_height = 64
```

Comment:
this breaks a number of checks, so you can't do it

Reply:
Also why the hell would you set these in update_icon